client/lib/purchases/index.js

```diff
@@ -130,6 +131,20 @@ function getSubscriptionEndDate( purchase ) {
 	return moment( purchase.expiryDate ).locale( localeSlug ).format( 'LL' );
 }
 
+/**
+ * Returns a purchase term label (i.e. "every month", "every year", "every two years").
+ *
+ * @param {object} purchase The purchase
+ * @returns {string} The purchase's term label
```

Comment:
Both of the functions called by this function can return `undefined`, so this function can also return `undefined`.

However, since this function is used directly in a translation requiring a string, it's probably best to have it return a string as you say here. In that case, you'd need to either change each of this function's `return` statements to have a string fallback (eg: ` return getWhatever() ?? '';`) or change `getPlanBillingTermLabel` and `getProductBillingTermLabel` to always return strings.

Reply:
Thanks for pointing this out. I opted to keep it `string|undefined` and added logic to the plan status string in e28983c.